### PR TITLE
README update & Solidity NatSpec comments

### PR DESCRIPTION
**Pull request by tchataigner**
## Goal of this PR

The main goal of this PR was for me to dive in cryptographic block used in our Nova Verifier ported in Solidity.

To do so, I've refactored the `README` and tried to produce NAtSpec documentation for our Solidity files. 

## Changelog

- Fixed typo for `EqPolynomial.sol`: EqPolinomialLib -> EqPolynomialLib
- Added NatSpec comments in all `*.sol` files in `src`
- Updated repository `README.md` to present repository structure, along with description and state of feature branches
- Added `README` in `src/blocks`, `src/blocks/grumpkin`, `src/blocks/pasta` and `src/blocks/poseidon` to describe their content.

## Related issue

N/A

### CODE REVIEW

**Review by storojs72 — COMMENTED** [review]
Thanks for such a diligent work, @tchataigner!

I haven't yet finished reviewing all NatSpec comments - but so far I have two common remarks (not directly related to the work itself):
- For future PRs I would recommend to setup GPG commits [signing](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification)
- Your target branch is `cleanup-main`, which means it also picks some commits from there. I expect some more commits added from @samuelburnham (https://github.com/lurk-lab/solidity-verifier/pull/45), so eventually we will need to specifying the proper order of PRs to be merged

**Comment by tchataigner** [comment]
> * Your target branch is `cleanup-main`, which means it also picks some commits from there. I expect some more commits added from @samuelburnham ([ci: Fail e2e tests when PR base is `main` #45](https://github.com/lurk-lab/solidity-verifier/pull/45)), so eventually we will need to specifying the proper order of PRs to be merged

Yes! I guess the way to go would be merge @samuelburnham PR in #42 , i'll merge the commit in mine, there should be no conflict. And if we plan to merge #42 in `main` I can change the target branch of this PR.

**Review by storojs72 — COMMENTED** [review]
I finished reviewing NatSpec. Thanks again for this work!

One thing, that I actually missed in https://github.com/lurk-lab/solidity-verifier/pull/42 is moving Sponge library that supports optimised Poseidon. While reviewing this PR it has been discovered and the https://github.com/lurk-lab/solidity-verifier/pull/42/commits/512e8b50f33d94d6f4617cf91866a8277ef1888b has been added. So, it makes sense to probably add NatSpec to `NovaSpongeLib` library.

File: src/blocks/grumpkin/Grumpkin.sol [hunk 1]
```diff
@@ -178,6 +232,12 @@ library Grumpkin {
         return GrumpkinAffinePoint(x, y);
     }
 
+    /**
+     * @dev Performs scalar multiplication on the Bn256 curve.
```
[the remark this answers is not in the document]
Comment (tchataigner) [reply src/blocks/grumpkin/Grumpkin.sol:236]: Sorry, copy/paste mistake

**Review by storojs72 — APPROVED** [review]

**Pull request merged**